lib/model/typing_status.dart

```diff
@@ -20,7 +20,7 @@ class TypingStatus extends ChangeNotifier {
   Iterable<SendableNarrow> get debugActiveNarrows => _timerMapsByNarrow.keys;
 
   Iterable<int> typistIdsInNarrow(SendableNarrow narrow) =>
-    _timerMapsByNarrow[narrow]?.keys ?? [];
+    _timerMapsByNarrow[narrow]?.keys.where((id) => id != selfUserId) ?? [];
```

Comment:
Ah, in https://github.com/zulip/zulip-flutter/pull/790#discussion_r1692310354 I didn't just mean doing the filtering in the model code rather than the UI code — I meant doing so in the model data structure itself. That is, just never have an entry here where the key is the self-user.

That way we get to avoid the `where`. Which in turn gets us out of scanning through the whole list if there are more than two.